lib/compilers/client_credentials_exchange.js

```diff
@@ -3,12 +3,86 @@
 const Authz = require('../authorization');
 const Factory = require('./compilerFactory');
 
-module.exports = Factory.createCompiler(clientCredentialsExchangeHandler);
+class BaseError extends Error {
+    constructor(code, message) {
+        super(message);
+        this.code = code;
+        this.name = this.constructor.name;
+    }
+}
+
+class InvalidRequestError extends BaseError {
+    constructor(code, message) {
+        super(code, message);
+    }
+}
+
+class InvalidScopeError extends BaseError {
+    constructor(code, message) {
+        super(code, message);
+    }
+}
+
+class ServerError extends BaseError {
+    constructor(code, message) {
+        super(code, message);
+    }
+}
+
+const ERROR_TO_RESULT_MAP =  {
+    [InvalidRequestError.name]: (err) => ({
+        result: 'oauth_error',
```

Comment:
After reviewing https://github.com/auth0/auth0-ext/pull/111/files and saw how this is used, I think the name should match the name of the hook, that way, there would be some consistency/format for other hooks to follow. Something like below:
```suggestion
        result: 'client_credential_exchange_error',
```